credit.py

```diff
@@ -7,16 +7,27 @@
                     format='%(asctime)s %(message)s', 
                     filemode='w') 
 logger = logging.getLogger() 
-logger.setLevel(logging.ERROR) 
+logger.setLevel(logging.WARNING) 
 
 def load_credits(path_credit):
 
     try:
         with open(path_credit) as f:
             credit = json.load(f)
-            if type(credit) is not dict:
+            if not isinstance(credit, dict):
                 logger.error('Data in credentials.txt is not dictionary')
-                return None
+                raise TypeError
+            k = 0
+            v = 0
+            for key in credit:
+                if not isinstance(key, str):
+                    k += 1
+            for value in credit:
```

Comment:
`v` and `k` will always be the same here, you want to look at `credit.values()` or `credit.items()` to count both at the same time.

Reply:
Don't understand, why `v` and `k` will always be the same here?

Reply:
I agree that logic is the same, but we can get case, where number of keys and values, which are not string, is not the same.

Reply:
For example we have dictionary `d = {'foo': 'bar', 'baz': 0, 4: 'quuz', 5: 8, 'pass': 7}`, where 3 values and 1 keys are not string.

Reply:
You probably want to modify this code as it still does the same thing for `k` and `v`.